src/utils/validation.js

```diff
@@ -15,6 +15,13 @@ function courseTagValidate(tagValue, selectValue, options) {
   return ![...selectValue, ...options].some(x => x.label.toLowerCase() === tagValue.toLowerCase());
 }
 
+function emailValidate(emailValue, selectValue, options) {
+  // emailValue should only contain alphabets, numbers, hyphen, underscore, dot and @
+  if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/i.test(emailValue)) { return false; }
+  // disallow emails that have already been selected or are present in options(dropdown)
+  return ![...selectValue, ...options].some(x => x.label.toLowerCase() === emailValue.toLowerCase());
```

Comment:
This particular validation should be on component level. This util method should only focus on email text validation.

Reply:
I've followed here the same convention which is currently followed for the tags validation

- https://github.com/openedx/frontend-app-publisher/blob/master/src/utils/validation.js#L11-L16
- https://github.com/openedx/frontend-app-publisher/blob/master/src/components/EditCoursePage/EditCourseForm.jsx#L511-L512

Reply:
Right. Tags have a specific use-case. Email is generic, in my opinion.

Reply:
I don't find any benefit in moving it to the component level. Because we can reuse this function anywhere after importing it to that file